### PR TITLE
Builder: use OutputPaths for root names

### PR DESCRIPTION
**Pull request by Profpatsch**
This is the second part of https://github.com/target/lorri/pull/117 for easier review.

Contains https://github.com/target/lorri/pull/134

### CODE REVIEW

File: src/build_loop.rs [hunk 1]
```diff
@@ -106,23 +105,38 @@ impl<'a> BuildLoop<'a> {
...
+        )?;
+        let root_shell = roots.add(
+            // TODO: bad, this magic name is not known here (only in roots)
+            "attr-shell",
+            // TODO: add actual root (this is just the same as the shell_root
+            &build
+                .drvs
+                .get(0)
+                .expect("logged-evaluation.nix didn’t return a drv!"),
+        )?;
```
Comment (grahamc) [inline src/build_loop.rs]: Do we want to be rooting the same drv twice?

File: src/builder.rs [hunk 2]
```diff
@@ -53,71 +50,102 @@ pub fn run(root_nix_file: &NixFile, cas: &ContentAddressable) -> Result<Info, Er
...
-                    LogDatum::AttrDrv(name, drv) => {
-                        named_drvs.insert(name, drv);
-                    }
+                    LogDatum::ShellDrv(drv) => {
+                        // check whether we have seen this field before
+                        match output_paths.shell {
+                            None => { output_paths.shell = Some(DrvFile(drv)); }
+                            // programming error
+                            Some(DrvFile(old)) => panic!(
+                                "`lorri read` got attribute `{}` a second time, first path was {:?} and second {:?}",
```
Comment (grahamc) [inline src/builder.rs]: Is this possible?

File: src/build_loop.rs (hunk 1 above)
Comment (Profpatsch) [reply src/build_loop.rs]: Temporary, will go away once the `TODO`s are cleaned up.

File: src/builder.rs (hunk 2 above)
Comment (Profpatsch) [reply src/builder.rs]: Nope, that’s why it’s a programming error if it happens.

Comment (Profpatsch) [reply src/builder.rs]: outdated by rebase

File: src/build_loop.rs (hunk 1 above)
Comment (Profpatsch) [reply src/build_loop.rs]: outdated by rebase

**Pull request merged**